sync-with-upstream/entrypoint.sh

```diff
@@ -11,6 +11,22 @@ DOWNSTREAM_BRANCH="$3"
 [[ -n "${UPSTREAM_BRANCH}" ]] || exit 1
 [[ -n "${DOWNSTREAM_BRANCH}" ]] || exit 1
 
+#
+# If there is a PR already then this is a no-op.
+#
+# An alternative design would be to pile up more commits
+# from the upstream repo if there exist any into the
+# existing PR instead of exiting here. (e.g. we could
+# check the hash of the upstream HEAD and see if that
+# hash is part of the existing PR, and if it's not we'd
+# force push into the PR). We decided not to go with
+# that design due to the simplicity of the current one.
+#
+if [[ $(hub pr list -s open -h "projects/sync-with-upstream/${DOWNSTREAM_BRANCH}" | wc -l) -ge 1 ]]; then
+	echo "sync-with-upstream PR already exists"
```

Comment:
I don't consider this necessary, but I think it'd be beneficial if we could print the PR(s) here. E.g. perhaps add the output of `pr list` here (even though that adds a race between the check, and printing the error).

Reply:
Could probably use an array to avoid the race condition; e.g. something like this (using our `delphix/cloud-init` repo as the example here):
```
± mapfile -t prs < <(hub pr list)
± echo ${#prs[@]}
3
± for i in $(seq 0 ${#prs[@]}); do echo ${prs[$i]}; done
#23 DLPX-69865 IP configuration is lost if a MAC address changes
#22 DLPX-71546 [Backport of DLPX-69865 to 6.0.5.0] MAC address changes in HyperV platform
#20 Generate network config on first and subsequent boots

```

Reply:
I changed it to this as I don't really need to care about the contents of the output, rather than its existense:
```
SYNC_PR=$(hub pr list -s open -h "projects/sync-with-upstream/${DOWNSTREAM_BRANCH}")
if [[ -n "${SYNC_PR}" ]]; then
	echo "sync-with-upstream PR already exists: ${SYNC_PR}"
	exit 0
fi
```
Are you ok with me pushing this?